proxy.go

```diff
@@ -352,8 +356,14 @@ func (p *Proxy) handle(ctx *Context, conn net.Conn, brw *bufio.ReadWriter) error
 				proxyutil.Warning(res.Header, err)
 			}
 
-			res.Write(brw)
-			return brw.Flush()
+			if err := res.Write(brw); err != nil {
+				log.Debugf("martian: got error while writing response back to client: %v", err)
```

Comment:
This is the only spot that returns the error potentially, so I was thinking that it could be up to the caller to decide if this was an error or not. I believe currently it is always a real error though. I'll change it to error.